### ISSUE TITLE
A recorded macro with "Find Next" and then "Run until the end of file" crashes Notepad++

### ISSUE THREAD

**Issue by BdR76** [notepad-plus-plus/notepad-plus-plus#13342]
### Description of the Issue
When you record a macro that includes a step with "Find Next", and you select "Run a Macro Multiple times" with the option "Run until the end of file", then it will get stuck in a loop and Notepad++ freezes.

### Steps to Reproduce the Issue
Example file

          <mappingPart>
            <source column="v_0" />
            <destination observationCode="demin_leefstijl_0" />
          </mappingPart>
          <mappingPart>
            <source column="v_1" />
            <destination observationCode="demin_leefstijl_1" />
          </mappingPart>
          <mappingPart>
            <source column="v_2" />
            <destination observationCode="demin_leefstijl_2" />
          </mappingPart>

1. Type the text `pivotResultColumn="data" />`, select & copy/cut it so it's on the clipboard
2. Go to beginning of the example file
3. Go to menu `Macro > Start recording`
4. Ctrl+F to find next `<source column="v` to move the cursor to next instance
5. Press `End` key to go to end of that line
6. select last two characters `/>`
7. Paste `pivotResultColumn="data" />`
8. Go to menu `Macro > Stop recording`
9. Now go to `Macro > Run a Macro Multiple times`
10. Select `Run until the end of file` and press OK.

### Expected Behavior
The macro should run until the end of the file, so break/stop when "Find Next" wraps around to the beginning of the file, or alternatively when "Find Next" yields no more results.

### Actual Behavior
The macro runs indefinitely and Notepad++ freezes

### Debug Information
Notepad++ v8.4.8   (64-bit)
Build time : Dec 24 2022 - 19:40:27
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows Server 2019 Standard (64-bit) 
OS Version : 1809
OS Build : 17763.3887
Current ANSI codepage : 1252
Plugins : 
    ComparePlugin (2.0.2)
    CSVLint (0.4.6.2)
    CsvQuery (1.2.8)
    HexEditor (0.9.12)
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    NPPJSONViewer (1.40)
    NppMarkdownPanel (0.5)
    PoorMansTSqlFormatterNppPlugin (1.6.13.31508)
    RandomValuesNppPlugin (0.2.1)
    XMLTools (3.1.1.13)

fyi, I encountered this while trying to manually edit a large XML file, and paste the `pivotResultColumn` property to the end of each `source` tag, but only when column name starts with the letter `v`.

I've found a work-around which is to first `Find > Count` to see how many instances the Find Next will get, for example count is 1389, and then select `Run 1389 times` instead of "until end of file".

Also, maybe this is somehow related to issue #12126 ?

**Comment by molsonkiko** [comment]
I think there's an even easier way to make this happen:
1. create any multi-line file
2. start recording a macro
3. Ctrl+F for `^` in regex mode (search for the next start of line)
4. Click `Run until end of file`.
5. Click `Run until end of file` again (if the first click doesn't cause infinite loop).
6. you will get an infinite loop.

I'm like 95% sure that the issue is somewhere in the code block that starts [here](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/9ee9046be4e088303939a20e356ec7427f59c52d/PowerEditor/src/NppBigSwitch.cpp#L1490) but I don't know C++ so I'm apprehensive about trying to fix it myself.

**Comment by BdR76** [comment]
I'm also not that familiar with C++ but your fix looks good. 

As far as I could tell this specific issue is caused somewhere in that if-statement at `// eof ?`, but it's all on one line and kind of hard to follow. I would change it so that each condition is on a new line, like in [this example](https://stackoverflow.com/a/29329481/1745616).
